File: utils/ngrams.py
import cPickle as pickle


def load_initial_ngrams():
    with open('data/initial_ngrams.pck', 'rb') as f:
        initial_ngrams = pickle.load(f)
        initial_ngrams[0] = ['']
        return initial_ngrams


def load_ngrams():
    with open('data/ngrams.pck', 'rb') as f:
        return pickle.load(f)

INITIAL_NGRAMS = load_initial_ngrams()
NGRAMS = load_ngrams()

File: go/src/cryptics/ngram_load_utils/ngrams.go
package ngram_load_utils

import (
	"encoding/gob"
	"fmt"
	"os"
)

// type Ngrams struct {
// 	Initial map[string]bool
// 	All     map[string]bool
// }

func LoadNgrams() map[int]map[string]bool {
	var ngrams map[int]map[string]bool
	raw_ngrams, err := os.Open("data/ngrams.gob")
	if err != nil {
		fmt.Println(err)
	}
	gob.NewDecoder(raw_ngrams).Decode(&ngrams)
	return ngrams
}

func LoadInitialNgrams() map[int]map[string]bool {
	var initial_ngrams map[int]map[string]bool
	init_ngrams_file, err := os.Open("data/initial_ngrams.gob")
	if err != nil {
		fmt.Println(err)
	}
	gob.NewDecoder(init_ngrams_file).Decode(&initial_ngrams)
	return initial_ngrams
}

var NGRAMS = LoadNgrams()
var INITIAL_NGRAMS = LoadInitialNgrams()
